### ISSUE TITLE
docs: showcase Laplacian smoothing and refresh the curvature figure

### ISSUE THREAD

**Issue by KhoiDOO** [KhoiDOO/conquer3d#59]
The Showcase gallery has no entry for the mesh smoothing shipped in #58 / PR #57, and the
curvature figure predates the multi-mode curvature rewrite, so it shows a narrower set of
fields than the library now computes.

## Scope

**Mesh section, new figure — Laplacian smoothing**

StanfordBunny (34,834 vertices, open: 223 boundary edges) smoothed for 10, 50 and 100
iterations with `damping=0.5`, `mode=1` (cotangent weights). Surfaces on the top row,
signed mean curvature beneath, all four curvature panels sharing one colour range taken
from the unsmoothed mesh so the field is comparable across the series.

**Mesh section, updated figure — curvature**

Mean curvature in all three modes plus Gaussian and both principal curvatures, one panel
each, signed fields on a diverging scale centred on zero.

**Compositor**

`compose.colormap` gains `symmetric` (centres a diverging map on zero, so a lopsided
signed field does not report the wrong sign) and `vmin`/`vmax` (pins the range so a series
of panels can share one scale). Both opt-in; the default path is pixel-identical to the
previous output.

## Why the series needed a shared scale

With per-panel normalisation, a field that is genuinely shrinking looks unchanged — each
panel renormalises to its own extremes. Pinning the range is what makes the smoothing
series legible as a progression rather than four unrelated pictures.

## Known caveat

The smoothing figure was rendered against an extension build predating the merged
implementation, so its numbers are provisional and it needs a re-render. Tracked in the PR.

## Verification

`docs/generate.py --strict` — 1159/1159 documented, all internal links resolve.

**Comment by KhoiDOO** [comment]
## API usage

```python
import torch
from conquer3d.data_structure import TriangleMesh
from conquer3d.io import load_mesh

mesh = load_mesh("StanfordBunny")          # 34,834 vertices, open (223 boundary edges)

# Laplacian smoothing, in place. Boundary vertices are pinned.
#   iterations : explicit Euler steps
#   damping    : step size per iteration
#   mode       : 0 uniform, 1 cotangent, 2 area-normalised
mesh.smooth(iterations=10, damping=0.5, mode=1)

# Curvature on the result. mode 0/1/2 select the mean-curvature estimator.
H  = mesh.get_mean_curvature(0)            # (V,) float32 CUDA, signed
k1, k2 = mesh.get_principal_curvatures()   # (V,), (V,)
```

Re-running on a fresh copy each time, since `smooth` mutates in place:

```python
base_v, faces = mesh.vertices.clone(), mesh.triangles.int().clone()
for steps in (10, 50, 100):
    m = TriangleMesh(base_v.clone().contiguous(), faces.clone().contiguous())
    m.smooth(iterations=steps, damping=0.5, mode=1)
```

## Advantages

- Runs entirely on the GPU with no host round-trip; vertices are mutated in place, so a
  sweep over iteration counts costs one clone per level rather than a readback.
- Cotangent weights (`mode=1`) approximate the Laplace-Beltrami operator, so smoothing
  follows the surface metric instead of the vertex valence.
- Boundary vertices are pinned, which keeps an open mesh from peeling back at its rim.

## Disadvantages and trade-offs

- **Shrinkage.** Explicit Laplacian smoothing loses volume. Measured surface-area loss on
  StanfordBunny: -3.5% / -9.0% / -12.8% at 10 / 50 / 100 iterations. If size must be
  preserved, this operator is the wrong one.
- **Mean curvature is not a monotone progress metric.** H scales as 1/length, so a smaller,
  smoother surface can report a *larger* |H|. Combined with shrinkage the series is
  non-monotonic.
- **Outlier sensitivity.** Cotangent weights blow up on degenerate triangles; near the ears
  and the open boundary these dominate an unclipped mean.
- Explicit Euler, so stability is bounded by `damping` x `iterations`.

## Empirical measurements

StanfordBunny, 34,834 vertices, `damping=0.5`, `mode=1`:

| iterations | mean \|H\| | max vertex shift | surface area |
|---|---|---|---|
| 0 | 6.860 | 0.00000 | +0.00% |
| 10 | 4.581 | 0.01648 | -3.47% |
| 50 | 5.785 | 0.03163 | -8.97% |
| 100 | 8.522 | 0.04313 | -12.83% |

> **These numbers are provisional.** They were produced by an extension build predating the
> implementation merged in #57, and must be regenerated after a rebuild before being quoted.

Docs build: 1159/1159 documented symbols, all internal links resolve.

## Technical insights

- **A shared colour range is required for a series.** Per-panel percentile normalisation
  makes a shrinking field look unchanged, because each panel rescales to its own extremes.
  `vmin`/`vmax` pin the range to the unsmoothed mesh so the four panels are comparable.
- **Zero must be centred on a diverging map.** With the range taken from the data, a
  lopsided signed field lands zero off-centre and the colours report the wrong sign — hence
  `symmetric`.
- **The mean disagrees with the picture, and the picture is right.** At 100 iterations the
  panel is visibly flattest yet reports the highest mean |H|. The rendered field is
  percentile-clipped; the mean is not. Median |H| tracks the bulk of the surface and is the
  honest summary statistic for this series.
- Reporting surface-area change beside the curvature is what turns a surprising number into
  a readable one: shrinkage plus 1/length scaling fully accounts for the rise.

**Comment by KhoiDOO** [comment]
Implemented and covered by PR #60.

Landed as five commits: the `colormap` range controls, the curvature figure rebuild, the
Laplacian smoothing showcase, the frozen v0.8.4 archive, and the v0.8.5 site regeneration.
Site verified at 1159/1159 with all internal links resolving.

Two follow-ups are deliberately **not** done here and should not be lost:

1. `fig-smoothing.webp` was rendered against an extension build predating the
   implementation merged in #57, so the published figure and every number in the comment
   above are provisional pending a re-render after a rebuild.
2. The bottom row reports mean |H|, which contradicts its own image at 100 iterations
   (flattest panel, highest value) because the displayed field is percentile-clipped while
   the mean is not. It should become median |H| at the same time as the re-render.
